TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Teleop24.java

```diff
@@ -56,21 +88,70 @@ public void loop(){
         telemetry.addData("Right Front Motor Encoder:", frontRight.getCurrentPosition());
 
         telemetry.addData("Right Back Motor Encoder:", rearRight.getCurrentPosition());
+        // Position of elevator arm motor encoder
+        telemetry.addData("Elevator Arm Position:", elevatorPivot.getCurrentPosition());
 
         double drive = (gamepad1.left_stick_y);//inverted???
         double strafe = (-gamepad1.left_stick_x);//inverted???
-        double turn = (gamepad1.right_stick_x);
+        double turn = (gamepad1.right_stick_x);//inverted???
+
+        boolean pivotUp = (gamepad2.y);
+        boolean pivotReset = (gamepad2.a);
+
+        boolean closeGrip = (gamepad2.left_bumper);
+        boolean openGrip = (gamepad2.right_bumper);
+
+        /**
+         </>his is some really janky math that someone implemented back in 2021, but hey, if it works, ¯\_(ツ)_/¯
+         drive = drive * drive * Math.signum(drive);
+         strafe = strafe * strafe * Math.signum(strafe);
+         turn = turn * turn * Math.signum(turn);
+         **/
+
+
+        //slightly less janky math that could work
+
+        drive = drive * Math.abs(drive);
+        strafe = strafe * Math.abs(strafe);
+        turn = turn * Math.abs(turn);
 
-        // This is some really janky math that someone implemented back in 2021, but hey, if it works, ¯\_(ツ)_/¯
-        drive = drive * drive * Math.signum(drive);
-        strafe = strafe * strafe * Math.signum(strafe);
-        turn = turn * turn * Math.signum(turn);
 
         // Allows for multiple functions to happen at once (eg. drive & strafe, turn & drive, etc)
         double denominator = Math.max(Math.abs(drive) + Math.abs(strafe) + Math.abs(turn), 1);
         frontLeft.setPower(maxSpeed*(drive - strafe + turn)/denominator);
         frontRight.setPower(maxSpeed*(drive - strafe - turn)/denominator);
         rearLeft.setPower(maxSpeed*(drive + strafe + turn)/denominator);
         rearRight.setPower(maxSpeed*(drive + strafe - turn)/denominator);
+
+        //PIVOT CONTROLS
+        // Move up to scoring position
+        if(pivotUp ) {
+            moveToPos(elevatorPivotUpSpeed, pivotTarget);
+            // May need to add a timeout here as it appears to take a while to stop.
+        }
+
+        // Move down to intake position
+        if(pivotReset ) {
+            moveToPos(elevatorPivotDownSpeed,pivotHome + 20);
+            moveToPos(elevatorPivotCrawlSpeed,pivotHome);

```

Comment:
Note that moveToPos is not blocking. Meaning it will not wait to reach the target position before returning. This means the first call to moveToPos has no effect because the second call will override the settings. It will move directly to pivotHome at elevatorPivotCrawlSpeed.

The fact that it doesn't block is actually a good thing. Otherwise you would be locked out of driving while the code waited for the arm to move.